src/components/SecretsModal/Annotations.js

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { TextInput } from 'carbon-components-react';
+import Add from '@carbon/icons-react/lib/add--alt/24';
+import Remove from '@carbon/icons-react/lib/misuse/24';
+import './SecretsModal.scss';
+
+const Annotations = props => {
+  const {
+    annotations,
+    handleChange,
+    handleAdd,
+    handleRemove,
+    invalidFields
+  } = props;
+
+  const annotationFields = annotations.map((annotation, index) => {
+    return (
+      <div className="annotationRow" key={`annotationRow(${index.toString()})`}>
```

Comment:
`index` should not be used for the key, instead you should use some attribute of the data that's unique, the label for example

Reply:
if the label or value for that annotation is used then when the user changes one of those fields it will unfocus after one character change which is quite annoying. ill see what I can do